packages/ethernaut-interact/src/tasks/calldata.js

```diff
@@ -0,0 +1,71 @@
+const { ethers } = require('ethers')
+const types = require('ethernaut-common/src/validation/types')
+const output = require('ethernaut-common/src/ui/output')
+const EthernautCliError = require('ethernaut-common/src/error/error')
+const abi = require('../../../ethernaut-interact-ui/src/suggest/abi')
+
+require('../scopes/interact')
+  .task(
+    'calldata',
+    'Decodes calldata of a provided one, or searches for it from a given tx id',
+  )
+  .addOptionalPositionalParam(
+    'calldata',
+    'The calldata to decode',
+    undefined,
+    types.string,
+  )
+  .addOptionalPositionalParam(
+    'transactionId',
+    'The transaction to lookup',
+    undefined,
+    types.bytes32,
+  )
+  .addPositionalParam(
+    'abi',
+    'The ABI of the contract to decode the calldata',
+    undefined,
+    types.array,
+  )
+  .addPositionalParam(
+    'filter',
+    'Some text to filter the list of known ABIs. Leave empty to list all. Results will be ordered by similarity to the filter text, so the first result will be the best match.',
+    undefined,
+    types.string,
+  )
```

Comment:
These two should not be part of the interface of the task.

Take a look at how the interact extension modifies its base plugin here:
https://github.com/theethernaut/ethernaut-cli/blob/main/packages/ethernaut-interact-ui/src/index.js#L19-L20

The base task doesnt know anything about the suggestion or custom prompt. Instead, the extension intercepts the task and assigns either a "suggest" or "prompt" property to the parameter.